nncf/onnx/graph/model_transformer.py

```diff
@@ -68,166 +99,101 @@ def transform(self, transformation_layout: TransformationLayout) -> onnx.ModelPr
+            target_edge_name = self._get_pre_post_layer_target_edge(port_id, node_name, transform_type, onnx_graph)
+            if target_edge_name is None:
+                raise RuntimeError('Could not find the edge corresponding to node {}'.format(node_name))
+            extra_model_outputs.append(target_edge_name)
+        return ONNXModelTransformer._insert_outputs(self._model, outputs=[*extra_model_outputs, *model_outputs])
+
+    @staticmethod
+    def _insert_outputs(model: onnx.ModelProto, outputs: List[str]) -> onnx.ModelProto:
         """
-        Takes a model and adds outputs based on the list of edge names to collect data from
+        Creates a new model as a copy of provided model with additional outputs.
 
-        :param model: *ONNX* model
-        :param outputs: edge names to collect data from
-        :return: modified model
+        :param model: Model of which copy will be created.
+        :param outputs: Edge names which are outputs.
+        :return: New model with provided outputs.
         """
-        if outputs is None:
-            raise RuntimeError("Parameter outputs cannot be None.")
         onnx_graph = ONNXGraph(model)
-        var_out = []
-        for out in outputs:
+        model_outputs = []
+        for output in outputs:
             # shape should be None; if you place not None, some models will have inference problems (e.g. Mask RCNN)
-            type_proto = onnx.helper.make_tensor_type_proto(onnx_graph.get_edge_dtype(out),
-                                                            shape=None)
-            value_info = onnx.helper.make_value_info(
-                name=out, type_proto=type_proto)
-            var_out.append(value_info)
-
-        graph = onnx.helper.make_graph(nodes=model.graph.node,
-                                       name=model.graph.name,
-                                       inputs=model.graph.input,
-                                       outputs=var_out,
-                                       initializer=model.graph.initializer,
-                                       value_info=model.graph.value_info)
-        onnx_model = onnx.helper.make_model(graph,
-                                            ir_version=model.ir_version,
-                                            producer_name=model.producer_name,
-                                            producer_version=model.producer_version,
-                                            domain=model.domain,
-                                            model_version=model.model_version,
-                                            doc_string=model.doc_string)
-        if len(model.metadata_props) > 0:
+            type_proto = onnx.helper.make_tensor_type_proto(onnx_graph.get_edge_dtype(output), shape=None)
+            model_outputs.append(onnx.helper.make_value_info(name=output, type_proto=type_proto))
+
+        graph = onnx.helper.make_graph(nodes=model.graph.node, name=model.graph.name,
```

Comment:
Make the formatting better, please.

Reply:
Below. Looks more readable, isn't it?

Reply:
It will double the necessary lines for the function. May we keep it as is now?